samples/net/socket_echo/prj_qemu_x86.conf

```diff
@@ -0,0 +1,26 @@
+# General config
+CONFIG_NEWLIB_LIBC=y
+
+# Networking config
+CONFIG_NETWORKING=y
+CONFIG_NET_IPV4=y
+CONFIG_NET_IPV6=n
+CONFIG_NET_TCP=y
+CONFIG_NET_SOCKETS=y
+CONFIG_NET_SOCKETS_POSIX_NAMES=y
+
+# Network driver config
+CONFIG_NET_SLIP_TAP=y
+CONFIG_TEST_RANDOM_GENERATOR=y
+
+# Without CONFIG_NET_BUF_LOG printf() doesn't work
```

Comment:
I do not really understand how net_buf logging could affect printf(), could you elaborate the issue more.

BTW, typically I have this setting in my prj.conf files

CONFIG_NET_LOG=y
CONFIG_SYS_LOG_NET_LEVEL=2

and then setting relevant net debug options to "y" and increasing CONFIG_SYS_LOG_NET_LEVEL to 4, will then start to print debug output.